File: 6KYU/Take a Ten Minutes Walk/solution.cpp
#include<vector>

bool isValidWalk(std::vector<char> walk) {
  int y = 0; int x = 0;
  
  for(int i = 0; i < walk.size(); i++) {
    if(walk[i] == 'n') {
      y++;
    } else if(walk[i] == 's') {
      y--;
    } else if(walk[i] == 'e') {
      x++;
    } else if(walk[i] == 'w') {
      x--;
    }
  }
  return walk.size() == 10 && y == 0 && x == 0;
}

File: 6KYU/Take a Ten Minutes Walk/solution.rs
use std::collections::HashMap;

fn is_valid_walk(walk: &[char]) -> bool {
    let mut dirs = HashMap::from([
       ('w', 0), ('e', 0), ('n', 0), ('s', 0)]);
    
    for c in walk {
        if dirs.contains_key(c) {
            dirs.insert(*c, 1 + dirs[&c]);
        }
    }
    
    return dirs[&'w'] == dirs[&'e'] && dirs[&'n'] == dirs[&'s'] && walk.len() == 10;
}
